course/src/isometry.cpp

```diff
@@ -2,16 +2,234 @@
+  new_x = this->x_ - r_vector.x_;
+  new_y = this->y_ - r_vector.y_;
+  new_z = this->z_ - r_vector.z_;
+
+  return Vector3(new_x, new_y, new_z);
+}
+
+Vector3& Vector3::operator-=(const Vector3& r_vector) {
+  this->x_ = this->x_ - r_vector.x_;
+  this->y_ = this->y_ - r_vector.y_;
+  this->z_ = this->z_ - r_vector.z_;
+
+  return *this;
+}
+
+Vector3 Vector3::operator*(const double& int_value) const {
+  double new_x, new_y, new_z;
+  new_x = this->x_ * int_value;
+  new_y = this->y_ * int_value;
+  new_z = this->z_ * int_value;
+
+  return Vector3(new_x, new_y, new_z);
+}
+
+Vector3& Vector3::operator*=(const double& int_value) {
+  this->x_ = this->x_ * int_value;
+  this->y_ = this->y_ * int_value;
+  this->z_ = this->z_ * int_value;
+
+  return *this;
+}
+
+Vector3 Vector3::operator*(const Vector3& r_vector) const {
+  double new_x, new_y, new_z;
+  new_x = this->x_ * r_vector.x_;
+  new_y = this->y_ * r_vector.y_;
+  new_z = this->z_ * r_vector.z_;
+
+  return Vector3(new_x, new_y, new_z);
+}
+
+Vector3& Vector3::operator*=(const Vector3& r_vector) {
+  this->x_ = this->x_ * r_vector.x_;
+  this->y_ = this->y_ * r_vector.y_;
+  this->z_ = this->z_ * r_vector.z_;
+
+  return *this;
+}
+
+Vector3 Vector3::operator/ (const double& int_value) const {
+  double new_x, new_y, new_z;
+  new_x = this->x_ / int_value;
+  new_y = this->y_ / int_value;
+  new_z = this->z_ / int_value;
+
+  return Vector3(new_x, new_y, new_z);
+}
+
+Vector3& Vector3::operator/=(const double& v) {
+  this->x_ = this->x_ / v;
+  this->y_ = this->y_ / v;
+  this->z_ = this->z_ / v;
+
+  return *this;
+}
+
+Vector3 Vector3::operator/(const Vector3& r_vector) const {
+  double new_x, new_y, new_z;
+  new_x = this->x_ / r_vector.x_;
+  new_y = this->y_ / r_vector.y_;
+  new_z = this->z_ / r_vector.z_;
+
+  return Vector3(new_x, new_y, new_z);
+}
+
+Vector3& Vector3::operator/=(const Vector3& r_vector) {
+  this->x_ = this->x_ / r_vector.x_;
+  this->y_ = this->y_ / r_vector.y_;
+  this->z_ = this->z_ / r_vector.z_;
+
+  return *this;
+}
+
+double Vector3::dot(const Vector3& r_vector) const {
+  double result;
+  result = this->x_ * r_vector.x_ + this->y_ * r_vector.y_ +
+  this->z_ * r_vector.z_;
+  return result;
+}
+
+Vector3 Vector3::cross(const Vector3& r_vector) const {
+  double new_x, new_y, new_z;
+  new_x = this->y_ * r_vector.z_ - this->z_ * r_vector.y_;
+  new_y = this->z_ * r_vector.x_ - this->x_ * r_vector.z_;
+  new_z = this->x_ * r_vector.y_ - this->y_ * r_vector.x_;
+
+  return Vector3(new_x, new_y, new_z);
+}
+
+double Vector3::norm() const {
+  double aux_x, aux_y, aux_z, result;
+  aux_x = pow(this->x_ , 2);
+  aux_y = pow(this->y_ , 2);
+  aux_z = pow(this->z_ , 2);
+  result = sqrt(aux_x + aux_y + aux_z);
+  return result;
+}
+
+Vector3 operator*(const int i, const Vector3& r_vector) {
+  double new_x, new_y, new_z;
+  new_x = i * r_vector.x();
+  new_y = i * r_vector.y();
+  new_z = i * r_vector.z();
+  return Vector3(new_x, new_y, new_z);
```

Comment:
Better just change the order of the operands and exploit the reversed operator overload.
```
return r_vector * i;
```
The more copies of the same algorithm you add to your code, the more chances you'll make an error in one, and the longer that error will go unnoticed.

Reply:
Please change the variable name, `i` is for index counters.